macros/sql/pivot.sql

```diff
@@ -0,0 +1,62 @@
+{#
+Pivot values from rows to columns.
+
+Example:
+
+    Input:
+
+    | size | color |
+    |------+-------|
+    | S    | red   |
+    | S    | blue  |
+    | S    | red   |
+    | M    | red   |
+
+    select
+      size,
+      {{ pivot('size', ['red', 'blue']) }}
+    from <table>
+    group by size
+
+    Output:
+
+    | size | red | blue |
+    |------+-----+------|
+    | S    | 2   | 1    |
+    | M    | 1   | 0    |
+
+Arguments:
+    column: Column name, required
+    values: List of row values to turn into columns, required
+    alias: Whether to create column aliases, default is True
+    op: SQL aggregation function, default is sum
+    cmp: SQL value comparison, default is =
+    prefix: Column alias prefix, default is blank
+    postfix: Column alias postfix, default is blank
+    then_value: Value to use if comparison succeeds, default is 1
+    else_value: Value to use if comparison fails, default is 0
+#}
+
+{% macro pivot(column,
+               values,
+               alias=True,
+               op='sum',
```

Comment:
I think `agg` might be more clear than `op` here (indicating that the group-by is required), but I'm ok with either